Geometry_Engine/Query/FilterByNormal.cs

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using System;
+using System.ComponentModel;
+
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns a list of PlanarSurfaces where each surface Normal is within the tolerance angle of the given vector.")]
+        [Input("surfaces", "A list of PlanarSurfaces to filter by a given Normal vector.")]
+        [Input("vector", "The Vector to filter by.")]
+        [Input("tolerance", "Angle tolerance used in geometry processing, default set to BH.oM.Geometry.Tolerance.Angle.")]
+        [Output("matchingSurfaces", "A list of PlanarSurfaces where each surface Normal is within the tolerance angle of the given vector.")]
+        public static List<PlanarSurface> FilterByNormal(this List<PlanarSurface> surfaces, Vector vector, double tolerance = Tolerance.Angle)
+        {
+            List<PlanarSurface> matchingSurfaces = new List<PlanarSurface>();
+            foreach (PlanarSurface srf in surfaces)
+            {
+                if (Math.Abs(Query.Angle(vector, srf.Normal())) < tolerance)
```

Comment:
You could use https://github.com/BHoM/BHoM_Engine/blob/main/Geometry_Engine/Query/IsParallel.cs here - additionally, possibly worth considering adding a `bool` method input to specify if the opposite vectors should be treated as equal?

Reply:
I did what you suggested, so the method's a little more flexible now. 

I'd like to have it a bit more generalised (applied to lists of like-objects), but as it's being driven by a pretty recurrent need of mine I'm happy with just the `PlanarSurface` version for now.